plugins/maciVoting/contexts/CoordinatorContext.tsx

```diff
@@ -0,0 +1,214 @@
+import { type KeyLike } from "crypto";
+import { EMode } from "@maci-protocol/sdk";
+import { PUB_COORDINATOR_SERVICE_URL, PUB_MACI_ADDRESS } from "@/constants";
+import { createContext, type ReactNode, useCallback, useMemo } from "react";
+import { hashMessage, toBytes } from "viem";
+import { usePublicClient, useSignMessage } from "wagmi";
+import {
+  type CoordinatorServiceResult,
+  type GenerateResponse,
+  type SubmitResponse,
+  type CoordinatorContextType,
+} from "./types";
+import { encryptWithCoordinatorRSA } from "./auth";
+import { GenerateResponseSchema, SubmitResponseSchema } from "./schemas";
+
+const baseUrl = PUB_COORDINATOR_SERVICE_URL;
+const maciContractAddress = PUB_MACI_ADDRESS;
+
+export const CoordinatorContext = createContext<CoordinatorContextType | undefined>(undefined);
+
+export const CoordinatorProvider = ({ children }: { children: ReactNode }) => {
+  const publicClient = usePublicClient();
+  const { signMessageAsync } = useSignMessage();
+
+  const getPublicKey = async (): Promise<KeyLike> => {
```

Comment:
Let's use `useCallback` here and add it to the deps for other `useCallback`s

Reply:
Done